File: generate_share_image.py
#!/usr/bin/env python3
"""
小凡看见分享图片生成脚本
生成 1080x1080 的正方形分享图片
"""

import os
import subprocess
from PIL import Image, ImageDraw, ImageFont

def find_font():
    """
    查找系统中的中文字体
    """
    # macOS 字体路径列表
    font_paths = [
        "/System/Library/Fonts/PingFang.ttc",
        "/System/Library/Fonts/STHeiti Light.ttc",
        "/Library/Fonts/SimHei.ttf",
        "/Library/Fonts/Arial Unicode.ttf",
        "/System/Library/Fonts/Helvetica.ttc",
    ]

    for font_path in font_paths:
        if os.path.exists(font_path):
            print(f"📝 使用字体: {font_path}")
            return font_path

    # 如果找不到任何字体，尝试用 brew 安装中文字体
    print("⚠️  未找到中文字体，尝试安装...")
    try:
        subprocess.run(["brew", "install", "font-noto-sans-cjk"], check=True)
        return "/usr/local/opt/font-noto-sans-cjk/share/fonts/opentype/noto-cjk/NotoSansCJK-Regular.ttc"
    except:
        print("❌ 字体安装失败，使用默认字体（可能显示不正确）")
        return None

def generate_share_image():
    # 图片尺寸
    width = 1080
    height = 1080

    # 创建图片，背景颜色为蓝色
    img = Image.new('RGB', (width, height), color='#5B9FE3')
    draw = ImageDraw.Draw(img)

    # 创建蓝色渐变背景
    # 从浅蓝到深蓝
    for y in range(height):
        # 渐变色计算
        ratio = y / height
        r = int(91 - (91 - 61) * ratio)  # 91 -> 61
        g = int(159 - (159 - 123) * ratio)  # 159 -> 123
        b = int(227 - (227 - 199) * ratio)  # 227 -> 199

        color = (r, g, b)
        draw.line([(0, y), (width, y)], fill=color)

    # 添加微妙的网格纹理
    for x in range(0, width, 40):
        draw.line([(x, 0), (x, height)], fill=(255, 255, 255, 8))
    for y in range(0, height, 40):
        draw.line([(0, y), (width, y)], fill=(255, 255, 255, 8))

    # 查找并加载字体
    font_path = find_font()

    if font_path and os.path.exists(font_path):
        font_title = ImageFont.truetype(font_path, 50)
        font_main = ImageFont.truetype(font_path, 200)
    else:
        print("⚠️  使用默认字体，中文可能显示不正确")
        font_title = ImageFont.load_default()
        font_main = ImageFont.load_default()

    # 左上角标题
    title_text = "凡人晨读营-小凡看见"
    draw.text((50, 60), title_text, fill=(255, 255, 255), font=font_title)

    # 中心文字 - 分两行
    center_x = width / 2
    center_y = height / 2
    line_height = 220

    # 第一行：小凡
    text1 = "小凡"
    # 阴影效果
    draw.text((center_x + 3, center_y - line_height / 2 + 4), text1,
              fill=(0, 0, 0, 80), font=font_main, anchor="mm")
    # 主文字
    draw.text((center_x, center_y - line_height / 2), text1,
              fill=(255, 255, 255), font=font_main, anchor="mm")

    # 第二行：看见
    text2 = "看见"
    # 阴影效果
    draw.text((center_x + 3, center_y + line_height / 2 + 4), text2,
              fill=(0, 0, 0, 80), font=font_main, anchor="mm")
    # 主文字
    draw.text((center_x, center_y + line_height / 2), text2,
              fill=(255, 255, 255), font=font_main, anchor="mm")

    return img

def main():
    try:
        print("🎨 生成小凡看见分享图片...")

        img = generate_share_image()

        # 确定输出目录
        base_dir = os.path.dirname(os.path.abspath(__file__))
        assets_dir = os.path.join(base_dir, "miniprogram", "assets", "images")
        output_path = os.path.join(assets_dir, "share-insight.png")

        # 创建目录
        os.makedirs(assets_dir, exist_ok=True)

        # 保存图片
        img.save(output_path, "PNG")

        # 获取文件大小
        file_size = os.path.getsize(output_path) / 1024

        print(f"✅ 分享图片已生成: {output_path}")
        print(f"📐 尺寸: 1080x1080 px")
        print(f"💾 文件大小: {file_size:.2f} KB")

    except Exception as e:
        print(f"❌ 生成失败: {str(e)}")
        import traceback
        traceback.print_exc()
        exit(1)

if __name__ == "__main__":
    main()

File: generate-share-image.js
#!/usr/bin/env node

/**
 * 小凡看见分享图片生成脚本
 * 生成 1080x1080 的正方形分享图片
 */

const fs = require('fs');
const path = require('path');

// 检查是否安装了 canvas 库
let Canvas, createCanvas;
try {
  Canvas = require('canvas');
  createCanvas = Canvas.createCanvas;
} catch (err) {
  console.error('❌ 缺少 canvas 库');
  console.error('请先运行: npm install canvas');
  process.exit(1);
}

function generateShareImage() {
  const width = 1080;
  const height = 1080;

  const canvas = createCanvas(width, height);
  const ctx = canvas.getContext('2d');

  // 背景：蓝色渐变（与小程序配色接近）
  const gradient = ctx.createLinearGradient(0, 0, width, height);
  gradient.addColorStop(0, '#5B9FE3');      // 亮蓝色
  gradient.addColorStop(0.5, '#4A8FD8');    // 中蓝色
  gradient.addColorStop(1, '#3D7BC7');      // 深蓝色
  ctx.fillStyle = gradient;
  ctx.fillRect(0, 0, width, height);

  // 添加微妙的网格纹理效果
  ctx.strokeStyle = 'rgba(255, 255, 255, 0.03)';
  ctx.lineWidth = 1;
  for (let i = 0; i < width; i += 40) {
    ctx.beginPath();
    ctx.moveTo(i, 0);
    ctx.lineTo(i, height);
    ctx.stroke();
  }
  for (let i = 0; i < height; i += 40) {
    ctx.beginPath();
    ctx.moveTo(0, i);
    ctx.lineTo(width, i);
    ctx.stroke();
  }

  // 左上角：标题文字
  ctx.font = '26px -apple-system, BlinkMacSystemFont, "Segoe UI", sans-serif';
  ctx.fillStyle = 'rgba(255, 255, 255, 0.95)';
  ctx.textAlign = 'left';
  ctx.textBaseline = 'top';
  ctx.fillText('凡人晨读营-小凡看见', 50, 60);

  // 中心：大标题 "小凡看见" - 分两行显示
  ctx.font = 'bold 140px -apple-system, BlinkMacSystemFont, "Segoe UI", sans-serif';
  ctx.textAlign = 'center';
  ctx.textBaseline = 'middle';

  const centerY = height / 2;
  const lineHeight = 160;

  // 第一行：小凡
  // 阴影效果
  ctx.fillStyle = 'rgba(0, 0, 0, 0.15)';
  ctx.fillText('小凡', width / 2 + 2, centerY - lineHeight / 2 + 3);

  // 主文字
  ctx.fillStyle = 'rgba(255, 255, 255, 1)';
  ctx.fillText('小凡', width / 2, centerY - lineHeight / 2);

  // 第二行：看见
  // 阴影效果
  ctx.fillStyle = 'rgba(0, 0, 0, 0.15)';
  ctx.fillText('看见', width / 2 + 2, centerY + lineHeight / 2 + 3);

  // 主文字
  ctx.fillStyle = 'rgba(255, 255, 255, 1)';
  ctx.fillText('看见', width / 2, centerY + lineHeight / 2);

  return canvas;
}

// 主函数
async function main() {
  try {
    console.log('🎨 生成小凡看见分享图片...');

    const canvas = generateShareImage();
    const buffer = canvas.toBuffer('image/png');

    // 确定输出目录
    const assetsDir = path.join(__dirname, 'miniprogram', 'assets', 'images');
    const outputPath = path.join(assetsDir, 'share-insight.png');

    // 创建目录（如果不存在）
    if (!fs.existsSync(assetsDir)) {
      fs.mkdirSync(assetsDir, { recursive: true });
      console.log(`📁 创建目录: ${assetsDir}`);
    }

    // 保存图片
    fs.writeFileSync(outputPath, buffer);
    console.log(`✅ 分享图片已生成: ${outputPath}`);
    console.log(`📐 尺寸: 1080x1080 px`);
    console.log(`💾 文件大小: ${(buffer.length / 1024).toFixed(2)} KB`);

  } catch (error) {
    console.error('❌ 生成失败:', error.message);
    process.exit(1);
  }
}

main();
